Sources/NIOCore/EventLoopFuture.swift

```diff
@@ -1570,3 +1570,16 @@ public struct _NIOEventLoopFutureIdentifier: Hashable {
         return UInt(bitPattern: ObjectIdentifier(future)) ^ 0xbf15ca5d
     }
 }
+
+#if compiler(>=5.5)
+// EventLoopPromise is a reference type, but by its very nature is Sendable.
+extension EventLoopPromise: Sendable { }
+
+// EventLoopFuture is a reference type, but it is Sendable. However, we enforce
+// that by way of the guarantees of the EventLoop protocol, so the compiler cannot
+// check it.
+extension EventLoopFuture: @unchecked Sendable { }
```

Comment:
I think we should only make the EventLoopFuture sendable, IF the value is sendable as well. Users might have classes that they pass through Futures and those must be thread safe. Declaring the Future sendable by default might send the wrong impression.

Reply:
The `Future` is _always_ sendable: you cannot mutate it directly and all callbacks execute on the same thread.